### ISSUE TITLE
Compilation failure with avro4s-core 4.0.8

### ISSUE THREAD

**Issue by anilkumarmyla** [sksamuel/avro4s#648]
4.0.7 works fine, updating to 4.0.8 runs into the following compilation failure

```
17:36:38 [error] Symbol 'type shapeless.ops.FnFromProductInstances.Aux' is missing from the classpath.
17:36:38 [error] This symbol is required by 'value io.circe.generic.decoding.IncompleteDerivedDecoders.ffp'.
17:36:38 [error] Make sure that type Aux is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
17:36:38 [error] A full rebuild may help if 'IncompleteDerivedDecoders.class' was compiled against an incompatible version of shapeless.ops.FnFromProductInstances.
17:36:42 [error] one error found
17:36:42 [error] (Compile / compileIncremental) Compilation failed 
```

**Comment by chenharryhua** [comment]
looks like it is caused by shapeless 2.3.6. upgrade to 2.3.7 may fix it. I use scala 2.12.

**Comment by sksamuel** [comment]
Merged, releasing 4.0.9
